src/main/java/seedu/duke/command/MarkAsDoneCommand.java

```diff
@@ -28,17 +28,26 @@ public MarkAsDoneCommand(String description, Grading grade) {
 
     @Override
     public void execute(SemesterList semesterList, AvailableModulesList availableModulesList)
-            throws RuntimeException, StorageException, InputException {
+            throws RuntimeException, InputException, StorageException {
         markAsDoneCommand(semesterList);
         Ui.showDoneMessage();
         super.execute(semesterList, availableModulesList);
     }
 
+    /** Find selected module in sem and assign module with a grade.
+     * Increase user's number of completed module credit if the grade assigned to module is not F or CU.
+     * @param semesterList Semester List containing Semester Module Lists, which contains selected modules
+     * @throws RuntimeException throws except if module not found in semester
+     */
     private void markAsDoneCommand(SemesterList semesterList) throws RuntimeException {
         for (SemModulesList sem: semesterList) {
             for (SelectedModule module: sem) {
-                if (module.getName().equals(description) || module.getId().equals(description)) {
-                    if (grade != Grading.F && grade != Grading.CU && !module.getDone()) {
+                boolean isModuleName = module.getName().equals(description);
+                boolean isModuleId = module.getId().equals(description);
```

Comment:
same for here, can we use equalsignorecase?

Reply:
Same here